es/es-server/src/main/java/org/elasticsearch/common/io/stream/ByteBufferStreamInput.java

```diff
@@ -77,7 +77,7 @@ public void readBytes(byte[] b, int offset, int len) throws IOException {
     }
 
     @Override
-    public void reset() throws IOException {
+    public synchronized void reset() throws IOException {
```

Comment:
This looks like it might be a false alert? I don't see how adding synchronized to the method adds any safety, given that other `buffer` accesses are not synchronized.

Reply:
The lgtm alert complains that the method signature in the subclass overriding a superclass method is lacking the synchronized keyword from the superclass method.

Reply:
yes I get that it complains, but adding `synchronized` everywhere might affect performance negatively without benefit.

I think we should look at these cases more closely and consider adding suppression's for the cases if possible or whitelist them somehow.

Reply:
Looks like that the synchronized keyword is not part of the method signature anyway, so it is technically valid to override without adding it. I wonder if a SupressWarning annotation would do the job here.

Reply:
@SuppressWarnings("sync-override") seems to the way to.